[DOCS-9557] adding storage monitoring doc

rmshree opened the pull request:
<!-- *Note: Please remember to review the Datadog Documentation [Contribution Guidelines](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md) if you have not yet done so.* -->

### What does this PR do? What is the motivation?
<!-- A brief description of the change being made with this pull request. What is your motivation for the PR? -->

Add a new "private" setup guide for the Storage Monitoring product in Preview.

### Merge instructions

<!-- If you're waiting for a release or there are other considerations that you want us to be aware of, list them here. -->

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### Additional notes
<!-- Anything else we should know when reviewing?-->

<!-- Previewing the PR: Assuming you are a Datadog employee and named your branch `<yourname>/<description>`, a preview build will run and links to the preview output will be auto-generated and posted in the PR comments. The links will 404 until the preview build is finished running. -->

buraizu commented:
DOCS-9557 has been created for review

buraizu commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,146 @@
+#### Step 3: Configure Inventory Generation
+
+For each bucket you want to monitor:
+  1. Go to the AWS S3 Console
```
```suggestion
  1. Go to the [Amazon S3 buckets page][5] in the AWS console, and select the bucket.
```

buraizu commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,146 @@
+
+For each bucket you want to monitor:
+  1. Go to the AWS S3 Console
+  2. Navigate to the bucket's Management tab
```
```suggestion
  2. Navigate to the bucket's **Management** tab.
```

buraizu commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,146 @@
+For each bucket you want to monitor:
+  1. Go to the AWS S3 Console
+  2. Navigate to the bucket's Management tab
+  3. Click Create Inventory Configuration
```
```suggestion
  3. Click **Create inventory configuration**.
```

buraizu commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,146 @@
+    - Set a configuration name
+    - (Optional) Specify a source bucket prefix
+    - We also advise picking “Current Versions Only”
+    - In select destination bucket, pick the newly created destination bucket. Example, if bucket is named destination-bucket `s3://your-destination-bucket`
```
```suggestion
    - **Destination**: select the destination bucket. For example, if the bucket is named `destination-bucket`, enter `s3://your-destination-bucket`.
```

buraizu commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,146 @@
+    - (Optional) Specify a source bucket prefix
+    - We also advise picking “Current Versions Only”
+    - In select destination bucket, pick the newly created destination bucket. Example, if bucket is named destination-bucket `s3://your-destination-bucket`
+    - If you wish to have a prefix on the destination bucket, please define it and let us know
```
```suggestion
       **Note**: If you want to use a prefix on the destination bucket, add this as well  
```

buraizu commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,146 @@
+    - We also advise picking “Current Versions Only”
+    - In select destination bucket, pick the newly created destination bucket. Example, if bucket is named destination-bucket `s3://your-destination-bucket`
+    - If you wish to have a prefix on the destination bucket, please define it and let us know
+    - Choose frequency: Daily (recommended) or Weekly [Note that this setting determines how often your prefix level metrics will be updated in Datadog]
```
```suggestion
    - **Frequency**: Datadog recommends choosing **Daily**. This setting determines how often your prefix-level metrics are updated in Datadog  
```

buraizu commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,146 @@
+    - In select destination bucket, pick the newly created destination bucket. Example, if bucket is named destination-bucket `s3://your-destination-bucket`
+    - If you wish to have a prefix on the destination bucket, please define it and let us know
+    - Choose frequency: Daily (recommended) or Weekly [Note that this setting determines how often your prefix level metrics will be updated in Datadog]
+    - Format: CSV
```
```suggestion
    - **Output format**: CSV  
```

buraizu commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,146 @@
+    - Format: CSV
+    - Enable inventory generation
+    - Do not specify encryption
+    - Select required metadata fields:
```
```suggestion
    - Select the following **Additional metadata fields**:  
```

buraizu commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,146 @@
+
+#### Post-Setup Steps
+
+After the above steps, please reach out on #storage-monitoring with the following information:
+
+1. Name of the destination bucket where inventories are stored
+2. Prefix where the inventory files will be stored
+2. Region of the destination bucket
+3. AWS Account ID in which the bucket is defined
+4. Datadog Role Name that has the permissions to read objects in destination bucket
+4. Datadog Org ID
```
```suggestion
After completing the above steps, reach out on the `#storage-monitoring` Slack channel with the following information:

1. Name of the destination bucket where inventories are stored.
2. Prefix to store the inventory files.
3. Region of the destination bucket.
4. AWS account ID containing the bucket.
5. Datadog role name that has the permissions to read objects in destination bucket.
6. Datadog org ID.
```

buraizu commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,146 @@
+### Support
+If you encounter any issues or need assistance:
+- Verify all permissions are correctly configured
+- Contact us at mahashree.rajendran@datadoghq.com with your bucket details, AWS account ID and Datadog Org ID
```
Please correct me if I'm wrong, but I think Datadog org ID isn't something that's readily available -- as far as I know, a user has to send a request to [this API endpoint](https://docs.datadoghq.com/api/latest/users/#get-user-details) and get the org ID from the returned data. Unless you're aware of another way for users to find their org ID, this is something we should probably provide guidance for

buraizu reviewed (changes requested):
Thanks for the PR! In addition to the edits requested:

- Image files are not rendering in their current location. They will need to be moved to `static/images/integrations/guide/storage_monitoring/`. Note that the `storage_monitoring` folder will need to be created first
- For the `cloud-inventory-policies-cfn.yaml` instructions, no information is provided for the **Configure stack options** screen. Is this intentional?

Please let me know if you have any feedback or questions!

rmshree commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,146 @@
+
+This template configures your existing S3 bucket to generate inventory reports, which Datadog will use to generate detailed metrics about your prefixes.
+
+1. Download the source-bucket-inventory-cfn.yaml template
```
Added the link in a new commit

rmshree commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,146 @@
+
+4. Verify parameters and launch stack
+
+{{< img src="integrations/guide/acknowledge_bucket_policy.png" alt="Verify and launch stack" responsive="true" style="width:60%;" >}}
```
The screenshot was partial view of the screen. I actually removed the screenshot because the bullet explains it pretty well.

rmshree commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,146 @@
+
+#### Post-Setup Steps
+
+After completing the CloudFormation setup, reach out with the following information:
```
We're working on creating an email handle but putting in my email in here until that's ready!

buraizu commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,159 @@
+   - Wait for the first inventory report to generate (up to 24 hours for daily inventories)
+   - Check the destination bucket for inventory files
+   - Confirm the Datadog integration can access the files:
+   - Navigate to **Infrastructure -> Resource Catalog -> Monitoring -> S3 Buckets -> Installation Recommendations** to see if the bucket you configured is showing in the list
```
```suggestion
       - Navigate to **Infrastructure -> Resource Catalog -> Monitoring -> S3 Buckets -> Installation Recommendations** to see if the bucket you configured is showing in the list
```

buraizu reviewed (changes requested):
This review:
- Fixes image addresses
- Moves links within tabs (each tab needs to contain its own set of links, apologies for my mistaken suggestions in the first review)
- Updates image widths to `90%`, because currently the text on them is too small to read without expanding the image
- Moves your email address out of the page and into the list of reference links, since our guidance on [PII in public content](https://datadoghq.atlassian.net/wiki/spaces/ISEC/pages/3013411411/Guidance+for+PII+in+Public+Blogs+and+Documentation) requests that these not be added to the docs. With that said, could users just as well reach out on #storage-monitoring instead, as mentioned on line `127`? If so, that would likely be the ideal approach.

Also, I'm guessing that it's intentional to **not** add a link to this page to the [integration guides page](https://docs.datadoghq.com/integrations/guide/), but let me know if that's incorrect

rmshree commented:
Thank you Bryce! Yes it's intentional that we've not added to the [integration guides page](https://docs.datadoghq.com/integrations/guide/) because we will be sharing this as a "private" link to interested customers and this page shouldn't be visible in search or lists.
I've also removed my personal email and replaced it with a common mailing list. I've also removed the slack handle.

urseberry commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,163 @@
+4. Click **Choose file** and select the `cloud-inventory-policies-cfn.yaml` file, then click **Next**.
+5. Fill in the required parameters:
+   - **DatadogIntegrationRole**: Your Datadog AWS integration role name
+   - **DestinationBucketName**: The name of the bucket that to receive your inventory files. **Note**: You must only use one destination bucket for all inventory files generated in an AWS account.
```
```suggestion
   - **DestinationBucketName**: The name of the bucket that receives your inventory files. **Note**: You must only use one destination bucket for all inventory files generated in an AWS account.
```

urseberry commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,163 @@
+#### Post-setup steps
+
+After completing the CloudFormation setup, [reach out][105] with the following information:
+    1. Name of the destination bucket holding the inventory files.
+    2. Prefix where the files are stored in the destination bucket.
+    3. Name of the source bucket you want to monitor (the bucket producing inventory files).
+    4. AWS region of the destination bucket holding the inventory files.
+    5. AWS account ID containing the buckets.
+    6. Datadog org ID.
```
In the preview, this list isn't getting formatted correctly. Let's try removing the indentation and see if that fixes it.
```suggestion
1. Name of the destination bucket holding the inventory files.
2. Prefix where the files are stored in the destination bucket.
3. Name of the source bucket you want to monitor (the bucket producing inventory files).
4. AWS region of the destination bucket holding the inventory files.
5. AWS account ID containing the buckets.
6. Datadog org ID.
```

urseberry commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,163 @@
+{{% /tab %}}
+{{% tab "AWS Console" %}}
+
+To manually set up the required [Amazon S3 Inventory][2] and related configuration, follow these steps:
```
The link to Amazon S3 Inventory is broken in the preview.

urseberry commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,163 @@
+  1. Go to the [Amazon S3 buckets page][203] in the AWS console, and select the bucket.
+  2. Navigate to the bucket's **Management** tab.
+  3. Click **Create inventory configuration**.
+  4. Configure the following settings:
```
This formatting is broken in the preview. To make a nested list, each level of nesting needs to be indented by 4 spaces. Here it looks like you may have indented it only 2 sapces.

urseberry commented on content/en/integrations/guide/storage-monitoring-setup.md:
```diff
@@ -0,0 +1,163 @@
+          2. Last Modified
+          3. Storage Class
+
+**Note**: Review [AWS S3 pricing][204] for costs related to inventory generation.
```
```suggestion
**Note**: Review [Amazon S3 pricing][204] for costs related to inventory generation.
```

urseberry commented:
/merge
